mqtt/mqtt-policy/src/substituter.rs

```diff
@@ -0,0 +1,261 @@
+use mqtt_broker::auth::{Activity, Operation};
+use policy::{Request, Result, Substituter};
+
+#[allow(clippy::doc_markdown)]
+/// MQTT-specific implementation of `Substituter`. It replaces MQTT and IoT Hub specific variables:
+/// * `iot:identity`
+/// * `iot:device_id`
+/// * `iot:module_id`
+/// * `iot:client_id`
+/// * `iot:topic`
+#[derive(Debug)]
+pub struct MqttSubstituter {
+    device_id: String,
+}
+
+impl MqttSubstituter {
+    pub fn new(device_id: impl Into<String>) -> Self {
+        Self {
+            device_id: device_id.into(),
+        }
+    }
+
+    fn device_id(&self) -> &str {
+        &self.device_id
+    }
+
+    fn replace_variable(&self, value: &str, context: &Request<Activity>) -> String {
+        if let Some(context) = context.context() {
+            if let Some(variable) = extract_variable(value) {
+                return match variable {
+                    "{{mqtt:client_id}}" => replace(value, variable, context.client_id().as_str()),
+                    "{{iot:identity}}" => replace(
+                        value,
+                        variable,
+                        &context.client_info().auth_id().to_string(),
```

Comment:
AuthId::Anonymous.to_string() -> "*"
is it fine?

Reply:
Good question. It is fine for now, since we don't support Anonymous for now. I have a note to think about that later.